src/Djot/Blocks.hs

```diff
@@ -1127,12 +1126,11 @@ closeContainingSections lev = do
       closeContainingSections lev
     _ -> pure ()
 
--- TODO avoid detour through String
 toIdentifier :: ByteString -> ByteString
 toIdentifier bs =
   if null parts
      then "sec"
-     else strToUtf8 $ intercalate "-" parts
+     else B8.intercalate (B8.singleton '-') parts
  where
    isSym = (`elem` ("][~!@#$%^&*(){}`,.<>\\|=+/" :: [Char]))
-   parts = words $ map (\c -> if isSym c then ' ' else c) $ utf8ToStr bs
+   parts = B8.words $ B8.map (\c -> if isSym c then ' ' else c) bs
```

Comment:
There's a subtle bug here which I missed when I reviewed this before. It manifested itself in
https://github.com/jgm/pandoc/issues/9541

The problem is the use of B8.words.  This splits on characters that satisfy `isSpaceWord8`, defined here:
https://hackage.haskell.org/package/bytestring-0.12.1.0/docs/src/Data.ByteString.Internal.Type.html#isSpaceWord8
Note that 0xa0 counts satisfies isSpaceWord8, and this is what is causing things to break here.

We either need to ensure that we split only on space characters < 0x80, OR just revert this and go back to the original (which wasn't appreciably slower).

Reply:
Oh whoops! I'm not confident that I would be able to implement a performant and correct variant of `B8.words`, so I would say we should revert this. Sorry for the hassle!

Reply:
No worries, I didn't see this one either!